### DISCUSSION TITLE
RHVoice as a library

### DISCUSSION THREAD

**Discussion by ghost** [RHVoice/RHVoice#522] [Q&A]
First of all, Olga, you're a hero, big thanks to you and everyone involved.

I would like to know how to use RHVoice as a library, after it had been installed. I tried many ways but always get "undefined reference". I'm a C++ beginner so I might be missing something basic.

I'm using RHVoice on personal projects[0], the stack is Debian 10 and Qt 5. I've made a package[1] with a modified test.cpp[2], which has a -d option to work as interactive, speaking the words that come from stdin.

I spawn a RHVoice-test process, and send words to it. This is far from ideal, as it has no control over playback, pitch, voice, etc...

The ideal would be a Qt Text-to-speech plugin, and to build that I need to use RHVoice as a library.

[0] https://bitbucket.org/rubens-schmitz/
[1] https://bitbucket.org/rubens-schmitz/rhvoice/
[2] https://bitbucket.org/rubens-schmitz/rhvoice/src/master/test.cpp

**Comment by alex19EP** [comment]
hello. you can have a look at /usr/include/RHVoice.h
if rhvoice is installed.

**Reply by ghost** [reply]
I tried and gave the same error, "undefined reference". I created a repository so you can see what I am missing:

https://github.com/rubens-schmitz/QtTextToSpeech_RHVoice

**Reply by ghost** [reply]
I added LIBS += -L/usr/lib -lRHVoice to the .pro file and the error is gone)

What is the void* user_data?

**Comment by alex19EP** [comment]
> What is the void* user_data?

I think @Olga-Yakovleva can describe more about it.

**Comment by Olga-Yakovleva** [comment]
This is whatever the developer using the library wants it to be. If you 
need some custom information to be available in your callback functions, 
set the user data to the pointer to that custom structure. If you don't 
need anything like that, just set it to NULL.
